docs/sources/lids/_DRAFT_QueryFairness.md

```diff
@@ -0,0 +1,319 @@
+---
+title: "XXXX: Query fairness across users within tenants"
+description: "Improve the query scheduler to ensure fairness between individual users of individual tenants."
+---
+
+# XXXX: Query fairness across users within tenants
+
+
+**Author:** Christian Haudum (<christian.haudum@grafana.com>)
+
+**Date:** 02/2023
+
+**Sponsor(s):** @chaudum @owen-d
+
+**Type:** Feature
+
+**Status:** Draft
+
+**Related issues/PRs:**
+
+**Thread from [mailing list](https://groups.google.com/forum/#!forum/lokiproject):**
+
+---
+
+## Background
+
+The query scheduler (or short scheduler) is a component of Loki that distributes requests (sub-queries) from the query frontend (or short frontend) to the querier workers so that execution fairness between tenants can be guaranteed.
+
+By maintaining separate FIFO queues for each tenant and assigning the correct amount of querier workers to these queues, the scheduler takes care that a single tenant cannot compromise all other tenants' query capabilities.
+
+**Component diagram:**
+
+```plantuml
+skinparam componentStyle rectangle
+
+package Frontend {
+    () RoundTripper as QF
+}
+
+package Scheduler {
+    () FrontendLoop as FL
+
+    component RequestQueue {
+        [Tenant Queue] as T1
+        [Tenant Queue] as T2
+    }
+
+    () QuerierLoop as Q1
+    () QuerierLoop as Q2
+
+}
+
+QF -->> FL
+FL -->> T1
+FL -->> T2
+
+package Querier {
+    [Worker] as W1
+    [Worker] as W2
+}
+
+T1 -->> Q1
+Q1 -->> W1
+
+T2 -->> Q1
+T2 -->> Q2
+Q2 -->> W2
+```
+
+![component diagram](../scheduler-component-diagram.png)
+
+**Sequence diagram:**
+
+```plantuml
```

Comment:
PlantUML is not supported by GitHub but [Mermaid is](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/creating-diagrams#creating-mermaid-diagrams). Maybe you could try it out :slightly_smiling_face:

Reply:
The docs will be rendered with Hugo before they go to the website. That said, our documentation setup does not support Mermaid nor PlantUML yet (there are requests to add support). That's why I added the rendered images as well.

Reply:
Ah sorry. I thought we had it. Well, you could add it manually https://mermaid.js.org/intro/#mermaid-api. I think going with Mermaid would make reviewing in GitHub a little easier :slightly_smiling_face: